### PR TITLE
Cleanup maass form imports, fixes #122

### LINKED ISSUE #122: never do from something import *
```
…/lmfdb/characters/test_characters.py:15: SyntaxWarning: import * only allowed at module level
  def test_Heckemethods(self):
```

This is just one part, also on module level we should never just import *. That's a very good practice and allows for static code checking like `pyFlakes` to work better!

(if you do not have `pyFlakes`, install it and run it on your files …)

### PR DESCRIPTION
**Pull request by AndrewVSutherland**
This PR is a pyflakes cleanup of the modular_forms/maass_forms/* subtree of the LMFDB that removes the last remaining import *'s in the LMFDB codebase (thereby allowing us close #122).

It uncovered many bugs due to references to misspelled or never defined variables; these all appear to be in code branches that are not (typically) executed or in functions that are not called anywhere in the LMFDB (a disturbing number of them were in maass_forms_db.py which is used and is the source of a few open production issues).

Where it was possible to discern the intent, I fixed these bugs, and where it was not I commented out the broken code and added a FIXME comment.  I also commented out some code that requires SQL imports and a SQL database to run (this code has not been touched in 4 years and appears not to be used).

All tests runs successful (except the known test_galois_conjugates errors) and browsing through the pages under  http://127.0.0.1:37777/ModularForm/GL2/Q/Maass/ did not reveal any changes -- everything that works on www.lmfdb.org appears to still work, and everything that is broken on www.lmfdb.org is still broken (but hopefully will now be easier to fix).

Pages I checked include:
http://127.0.0.1:37777/ModularForm/GL2/Q/Maass/BrowseGraph/1/10/0/10/
http://127.0.0.1:37777/ModularForm/GL2/Q/Maass/4cb8503a58bca91458000032
http://127.0.0.1:37777/L/ModularForm/GL2/Q/Maass/4cb8503a58bca91458000032/
http://127.0.0.1:37777/L/ModularForm/GL2/Q/Maass/4cb8503a58bca91458000032/?download=lcalcfile
http://127.0.0.1:37777/ModularForm/GL2/Q/Maass/1/
http://127.0.0.1:37777/ModularForm/GL2/Q/Maass/4cb8503a58bca91458000031
http://127.0.0.1:37777/ModularForm/GL2/Q/Maass/BrowseGraph/100/250/0/2/
http://127.0.0.1:37777/ModularForm/GL2/Q/Maass/BrowseGraph/100/1000/0/1/
http://127.0.0.1:37777/ModularForm/GL2/Q/Maass/53234e85acf756120945d7d0
http://127.0.0.1:37777/L/ModularForm/GL2/Q/Maass/53234e85acf756120945d7d0/
http://127.0.0.1:37777/L/ModularForm/GL2/Q/Maass/53234e85acf756120945d7d0/?download=lcalcfile
http://127.0.0.1:37777/ModularForm/GL2/Q/Maass/997/

I also checked that prev/next and download links (both of them) on Maass form pages still work.

### CODE REVIEW

**Comment by JohnCremona** [comment]
I can look at this (again) but not until tomorrow.  It would be nice if one of the authors of the code could be pinged about it!

BTW, in using pyflakes on my own code I came across several instances of something which can lead to bugs which are very hard to track down otherwise, though are sometimes harmless:  the variables used in a list comprehension are *not* local to that as one might think!  So it you have defined (say) "E = EllipticCurve(...)" and need to use the value of E later, you really should not put, a few lines later, "jlist = [E.j_invariant() for E in Elist]".  But I think that the people reading this conversation already know that...

**Comment by AndrewVSutherland** [comment]
@JohnCremona thanks, tomorrow is fine.

Regarding pyflakes, yes it can definitely reveal some very subtle bugs, which is why I think our efforts to do a pyflakes cleanup of the entire codebase are well worth while (and we are not all that far from being done, in the process of cleaning up imports we have also cleaned up almost everything flagged by pyflakes).

The list comprehension variable scoping you mention is a particularly nasty "feature" of python that I have always viewed as a bug -- the python implementation team apparently agrees, since they actually fixed this in python 3.  But of course we are still on python 2.x and likely will be for a long time due to dependencies in sage.  All the more reason to run pyflakes on everything.

Actually, once we have cleaned up all the pyflakes issues, I would like to propose that we add pyflakes to requirements.txt and modify test.sh to include the line "find | grep "\.py$" | xargs pyflakes" to automatically run pyflakes on all directories being searched for test scripts.  This will keep new problems (including import *'s) from creeping into the code.

**Comment by JohnCremona** [comment]
I didn't know that about python 3 -- good!

I agree with adding the running of pyflakes as standard to test.sh.  One thing which would make this work slightly better would be to move all code which is not used in running the website (import scripts mainly, but there are some others) to a new place: we already have lmfdb/import_scripts (which has some but not all impost scripts) but I was thinking of a separate top-level directory alongside perfmon and data_mgt.

**Comment by AndrewVSutherland** [comment]
@JohnCremona I like the idea of pulling code that is not part of the website out of the lmfdb directory tree a lot (I would still suggest people don't use import * and do use pyflakes -- bugs in import scripts are not a good thing either -- but I would not be so worried about eval and sage_eval).

**Comment by edgarcosta** [comment]
+1 for this:
> One thing which would make this work slightly better would be to move all code which is not used in running the website (import scripts mainly, but there are some others) to a new place

If we clearly separate the what is used for the website and what is not used in the website, then we can slowly raise the standards  and easily "enforce" the new standards for all the code that is used for the website.
For example, enforcing no sage_eval or eval, etc...

**Comment by JohnCremona** [comment]
We should probably open a new issue for the topic of the last few comments -- put all code not used by the live database into a separate subdirectory (say "local") of the top-level, starting by moveing lmfdb/import_scripts to local/impost_scripts.

Before we start doingt his, I think we should acknoeledge the fact that there ahve been a vast number of mostly minor code changes happening, which is very likely to have the side-effect that anyone out there with branches they were working on will now find themselves having to merge with the current master and dealing with some conflicts.  Of course this is inevitable, but from a human point of view it is likely to result in delays in people working on useful things, at least until they know that this process has been completed!

**Comment by JohnCremona** [comment]
About 80% or even 90% of the changes here are instantly mergable, but for the remaining 10-20% it would be preferable for someone who has written or worked on this code before to look at the changes, especially as there are only 2 small tests in lmfdb/modular_forms/maass_forms/, so saying that all tests pass is not saying so much.
Who should asked to take a look?  Or shall we merge this now anyway -- I saw nothing obviously wrong?

**Comment by AndrewVSutherland** [comment]
@JohnCremona Most of the code was originally written by @lemurell but has not been modified by him in a long time, so I'm not sure how familiar he is with the code.  I agree with you about the test cases (in fact I will raise an issue about this), which is why I also did a fair amount of manually browsing to test (e.g. you might also want to click through the list of URLs I put in the initial description of the PR to).

If @lemurell wants (and has time) to look through this PR that would be great!  But I don't think we should wait to merge.  There are at least 4 significant outstanding issues in the Maass forms code that need to be fixed for Release 1.1 (#1803, #1812, #1820, #1821), and we can't really start to work on them until after this PR is merged (otherwise we are going to have all sorts of merge conflicts to deal with).

I realize there is some risk, but I don't think anything significant is going to break (and I will take responsibility for fixing it if it does).  There are already a lot of bugs in the Maass forms code (a few of which are fixed by this PR), and while  I can't guarantee this PR won't introduce any new problems, on balance I think it will fix more than it breaks and more importantly, it puts us in a much better position to fix things going forward (the current import *'s in the code hide a lot of issues).

In other words, I think it does meet our standard for merging by making things "less bad".

**Comment by JohnCremona** [comment]
OK!

**Comment by AndrewVSutherland** [comment]
Per https://github.com/LMFDB/lmfdb/pull/1819#issuecomment-233564612, I just opened #1837.
